### PR TITLE
New swipe function for Android devices

### PR DESCRIPTION
**Pull request by lamkovod**
Brand new rewriten swipeToText function for Android devices. 
Faster and more stable. Using UIAutomator capabilities.
Added new Cucumber Step.

Old swipeToText can be useful for iOS devices in future releases.

### CODE REVIEW

File: src/main/java/ru/sbtqa/tag/pagefactory/stepdefs/GenericStepDefs.java [hunk 1]
```diff
@@ -420,6 +421,18 @@ public void swipeToText(String direction, String text) throws SwipeException {
...
 
+    /**
+     * Swipe until text is visible for Android
+     *
+     * @param strategy contains or exact
+     * @param text text on page to swipe to
+     * @throws SwipeException if the text is not found
+     */
+    @And("ru.sbtqa.tag.pagefactory.swipeToTextAndroid")
+    public void swipeToTextAndroid(String strategy, String text) throws SwipeException {
```
Comment (kosteman) [inline src/main/java/ru/sbtqa/tag/pagefactory/stepdefs/GenericStepDefs.java]: ^user swipe to text \"([^\"]*)\" using match strategy \"([^\"]*)\"$ - first text, second strategy
swipeToTextAndroid(String strategy, String text) - first strategy, second text

will this match properly?

File: src/main/java/ru/sbtqa/tag/pagefactory/extensions/MobileExtension.java [hunk 2]
```diff
@@ -1,6 +1,9 @@
 package ru.sbtqa.tag.pagefactory.extensions;
 
 import java.util.List;
+import java.util.NoSuchElementException;
+
```
Comment (kosteman) [inline src/main/java/ru/sbtqa/tag/pagefactory/extensions/MobileExtension.java]: remove this newline

File: src/main/java/ru/sbtqa/tag/pagefactory/stepdefs/GenericStepDefs.java (hunk 1 above)
Comment (lamkovod) [reply src/main/java/ru/sbtqa/tag/pagefactory/stepdefs/GenericStepDefs.java]: Of course, my mistake.
I'll fix it

File: src/main/resources/i18n/ru/sbtqa/tag/pagefactory/stepdefs/GenericStepDefs/ru.properties [hunk 3]
```diff
@@ -21,6 +21,7 @@ ru.sbtqa.tag.pagefactory.goToPageByUrl=^(?:пользователь |он |)пе
 ru.sbtqa.tag.pagefactory.goToUrl=^(?:пользователь |он |)(?:переходит на|открывает) url "(.*?)"$
 ru.sbtqa.tag.pagefactory.reInitPage=^обновляем страницу$
 ru.sbtqa.tag.pagefactory.swipeToText=^пользователь свайпает экран \"([^\"]*)\" до текста \"([^\"]*)\"$
+ru.sbtqa.tag.pagefactory.swipeToTextAndroid=^пользователь свайпает до текста \"([^\"]*)\" используя \"([^\"]*)\"$
```
Comment (kosteman) [inline src/main/resources/i18n/ru/sbtqa/tag/pagefactory/stepdefs/GenericStepDefs/ru.properties]: add some refinement to this stepdef like
^пользователь свайпает до текста \"([^\"]*)\" используя стратегию \"([^\"]*)\"$

File: src/main/java/ru/sbtqa/tag/pagefactory/extensions/MobileExtension.java (hunk 2 above)
Comment (lamkovod) [reply src/main/java/ru/sbtqa/tag/pagefactory/extensions/MobileExtension.java]: Should I use selenium class instead?

File: src/main/java/ru/sbtqa/tag/pagefactory/extensions/MobileExtension.java [hunk 4]
```diff
@@ -170,4 +173,34 @@ public static void swipeToText(DirectionStrategy direction, String text, MatchSt
...
+     * Swipe until the text becomes visible
+     *
+     * UIAutomator for Android. Much faster and stable
+     *
+     * @param text text on page to swipe to
+     * @param strategy contains or exact
+     * @throws SwipeException if there is an error while swiping
+     */
+    public static void  swipeToText(MatchStrategy strategy, String text) throws SwipeException {
+        switch(strategy) {
```
Comment (clicman) [inline src/main/java/ru/sbtqa/tag/pagefactory/extensions/MobileExtension.java]: Use break at the end of all cases.
And add default break too.

**Pull request merged**